File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType7RecordCreateForm.cs
using System;
using System.IO;
using System.Windows.Forms;
using Neurotec.Biometrics.Standards;
using Neurotec.IO;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType7RecordCreateForm : ANRecordCreateForm
	{
		#region Public properties

		public ANType7RecordCreateForm()
		{
			InitializeComponent();

			isrResolutionEditBox.PpmValue = ANType1Record.MinScanningResolution;
			irResolutionEditBox.PpmValue = ANType1Record.MinScanningResolution;
		}

		#endregion

		#region Protected methods

		protected override ANRecord OnCreateRecord(ANTemplate template)
		{
			byte[] data = File.ReadAllBytes(tbImageDatePath.Text);
			NBuffer buffer = new NBuffer(data);

			ANType7Record record = new ANType7Record(ANTemplate.VersionCurrent, Idc)
			{
				Data = buffer
			};

			template.Records.Add(record);
			return record;
		}

		#endregion

		#region Private methods

		private void BtnImageDataClick(object sender, EventArgs e)
		{
			if (imageDataOpenFileDialog.ShowDialog() == DialogResult.OK)
			{
				tbImageDatePath.Text = imageDataOpenFileDialog.FileName;
			}
		}

		#endregion
	}
}

File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType7RecordCreateForm.vb
Imports System
Imports System.IO
Imports Neurotec.Biometrics.Standards
Imports Neurotec.IO

Namespace RecordCreateForms

	Partial Public Class ANType7RecordCreateForm
		Inherits ANRecordCreateForm
#Region "Public properties"

		Public Sub New()
			InitializeComponent()

			isrResolutionEditBox.PpmValue = ANType1Record.MinScanningResolution
			irResolutionEditBox.PpmValue = ANType1Record.MinScanningResolution
		End Sub

#End Region

#Region "Protected methods"

		Protected Overrides Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Dim data() As Byte = File.ReadAllBytes(tbImageDatePath.Text)
			Dim buffer As New NBuffer(data)

			Dim record As New ANType7Record(ANTemplate.VersionCurrent, Idc) With {.Data = buffer}

			template.Records.Add(record)
			Return record
		End Function

#End Region

#Region "Private methods"

		Private Sub BtnImageDataClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnBrowseImageData.Click
			If imageDataOpenFileDialog.ShowDialog() = Windows.Forms.DialogResult.OK Then
				tbImageDatePath.Text = imageDataOpenFileDialog.FileName
			End If
		End Sub

#End Region
	End Class
End Namespace
